Hack: Kiota doesn't clean up models

This code:

https://github.com/octokit/source-generator/blob/d1c29eea0c06262a42cc119b3ab0304f5e8e67ef/post-processors/go/main.go#L153-L170

is in place to remove models that Kiota partially optimizes away but not fully. Observe this log snippet from generation:

```
info: Kiota.Builder.KiotaBuilder[0]
      Removing unused model codeScanningAlertSeverity as it is not referenced by the client API surface
info: Kiota.Builder.KiotaBuilder[0]
      Removing unused model contentDirectory as it is not referenced by the client API surface
info: Kiota.Builder.KiotaBuilder[0]
      Removing unused model codeScanningAlertStateQuery as it is not referenced by the client API surface
```

However, when `removeModelsKiotaDoesNotCleanUp` is removed from generation, building the generated SDK errors with the following:

```
+ go build ./...
# github.com/octokit/kiota/orgs
orgs/item_code_scanning_alerts_request_builder.go:33:108: undefined: i158396662f32fe591e8faa247af18558546841dba91f24f5c824e11e34188830.CodeScanningAlertSeverity
orgs/item_code_scanning_alerts_request_builder.go:43:107: undefined: i158396662f32fe591e8faa247af18558546841dba91f24f5c824e11e34188830.CodeScanningAlertStateQuery
# github.com/octokit/kiota/repos
repos/item_item_code_scanning_alerts_request_builder.go:32:108: undefined: i158396662f32fe591e8faa247af18558546841dba91f24f5c824e11e34188830.CodeScanningAlertSeverity
repos/item_item_code_scanning_alerts_request_builder.go:42:107: undefined: i158396662f32fe591e8faa247af18558546841dba91f24f5c824e11e34188830.CodeScanningAlertStateQuery
```

We should take this issue to the Kiota team and see if they can help pinpoint what's going wrong.

--- a/.github/workflows/build-go.yml
+++ b/.github/workflows/build-go.yml
@@ -26,7 +26,7 @@ jobs:
           go-version: '1.21.4'
 
       - name: Install kiota
-        run: dotnet tool install --global Microsoft.OpenApi.Kiota --version 1.8.1
+        run: dotnet tool install --global Microsoft.OpenApi.Kiota --version 1.8.2
 
       - name: Clone the existing go-sdk
         run: |

--- a/post-processors/go/main.go
+++ b/post-processors/go/main.go
@@ -43,7 +43,6 @@ func run() error {
 
 		fileContents = fixImports(fileContents)
 		fileContents = fixCreateDateOnlyFromDiscriminatorValue(fileContents, file.Name())
-		fileContents = removeModelsKiotaDoesNotCleanUp(fileContents)
 		fileContents = dirtyHackForVersionsRequestBuilder(fileContents, file.Name())
 		fileContents = fixKiotaNonDeterminism(fileContents, file.Name())
 
@@ -187,25 +186,6 @@ func dirtyHackForVersionsRequestBuilder(inputFile string, fileName string) strin
     //     if v != nil {
     //         val[i] = *(v.(*i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.DateOnly))
     //     }`
-
-	if strings.Contains(inputFile, toReplace) {
-		inputFile = strings.ReplaceAll(inputFile, toReplace, replaceWith)
-	}
-
-	return inputFile
-}
-
-func removeModelsKiotaDoesNotCleanUp(inputFile string) string {
-	toReplace := `// If specified, only code scanning alerts with this severity will be returned.
-    SeverityAsCodeScanningAlertSeverity *i158396662f32fe591e8faa247af18558546841dba91f24f5c824e11e34188830.CodeScanningAlertSeverity ` + "`uriparametername:\"severity\"`"
-	replaceWith := ``
-
-	if strings.Contains(inputFile, toReplace) {
-		inputFile = strings.ReplaceAll(inputFile, toReplace, replaceWith)
-	}
-
-	toReplace = `// If specified, only code scanning alerts with this state will be returned.
-    StateAsCodeScanningAlertStateQuery *i158396662f32fe591e8faa247af18558546841dba91f24f5c824e11e34188830.CodeScanningAlertStateQuery ` + "`uriparametername:\"state\"`"
 
 	if strings.Contains(inputFile, toReplace) {
 		inputFile = strings.ReplaceAll(inputFile, toReplace, replaceWith)

--- a/scripts/install-tools.sh
+++ b/scripts/install-tools.sh
@@ -4,7 +4,7 @@
 KIOTA_TOOL_NAME="microsoft.openapi.kiota"
 # This version can only increase, never decrease. If you want to use a lower version, you need to uninstall the tool first.
 # dotnet tool uninstall Microsoft.OpenApi.Kiota --global
-KIOTA_TOOL_VERSION="1.8.1" 
+KIOTA_TOOL_VERSION="1.8.2"
 
 # Check if the tool is installed globally
 if dotnet tool list -g | grep -q $KIOTA_TOOL_NAME; then

